internal/coreinternal/timeutils/parser.go

```diff
@@ -0,0 +1,98 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package timeutils // import "github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal/timeutils"
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	strptime "github.com/observiq/ctimefmt"
+)
+
+func StrptimeToGo(layout string) (string, error) {
```

Comment:
What's the benefit of creating this function over just directly calling `strptime.ToNative`? Are we just trying to centralize all of the necessary time logic in this package? It just seems unnecessary.

Reply:
Thats fair.  I was too focused on how stanza was doing things and not what will be most useful in a general sense.  Replacing this with a ParseStrptime function for OTTL or any other component to use makes sense.